src/main/scala/ztis/recommender/RecommenderService.scala

```diff
@@ -61,7 +62,7 @@ class RecommenderService extends StrictLogging {
     usersWithData
   }
 
-  private def haveDataFor(userId: Int) = {
-    allUsers.contains(userId)
+  private def haveDataFor(userId: Option[Int]) = {
+    userId.isDefined && allUsers.contains(userId.get)
```

Comment:
Currently we are storing all users from the model in memory: 
private val allUsers: Set[Int] = model.userFeatures.map(_._1).collect().toSet
I don't think this will scale well when we will have many users inside Cassandra database. I would keep it as RDD or just check if user-service returns Some(_) or None.